packages/flutter/test/material/bottom_app_bar_test.dart

```diff
@@ -85,6 +85,40 @@ void main() {
     );
   }, skip: isBrowser); // https://github.com/flutter/flutter/issues/44572
 
+  testWidgets('Custom Padding', (WidgetTester tester) async {
+  const EdgeInsets customPadding = EdgeInsets.all(20);
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Builder(
+          builder: (BuildContext context) {
+            return const Scaffold(
+              bottomNavigationBar: BottomAppBar(
+                padding: customPadding,
+                child: Text('text'),
+              ), 
+            );
+          },
+        ),
+      ),
+    );
+
+    final BottomAppBar bottomAppBar = tester.widget(find.byType(BottomAppBar));
+    expect(bottomAppBar.padding, customPadding);
+    final Offset babStart = tester.getTopLeft(find.byType(BottomAppBar));
```

Comment:
Why not just test `tester.getRect()`. To ensure that the bounds of the component are predictable you might use `SizedBox(width: 300, height: 100)` as the child, instead of a Text widget.